psi4/src/psi4/libsapt_solver/exch-ind30.cc

```diff
@@ -314,5 +319,305 @@ double SAPT2p3::exch_ind30_3(double **sBS) {
+    omega_ar->add(BJK_ar);
+    omega_ar->scale(2);
+
+    std::shared_ptr<Matrix> omega_as(AJK_as->clone());
+    omega_as->zero();
+    omega_as->add(AJK_as);
+    omega_as->add(BJK_as);
+    omega_as->scale(2);
+
+    std::shared_ptr<Matrix> omega_br(AJK_br->clone());
+    omega_br->zero();
+    omega_br->add(AJK_br);
+    omega_br->add(BJK_br);
+    omega_br->scale(2);
+
+    std::shared_ptr<Matrix> omega_bs(AJK_bs->clone());
+    omega_bs->zero();
+    omega_bs->add(AJK_bs);
+    omega_bs->add(BJK_bs);
+    omega_bs->scale(2);
+
+    Cocc0AB.reset();
+    D_Ni_a.reset();
+    D_Ni_b.reset();
+    J_D_ia.reset();
+    K_D_ia.reset();
+    J_D_ib.reset();
+    K_D_ib.reset();
+    AJK.reset();
+    BJK.reset();
+    AJK_ar.reset();
+    AJK_as.reset();
+    AJK_br.reset();
+    AJK_bs.reset();
+    BJK_ar.reset();
+    BJK_as.reset();
+    BJK_br.reset();
+    BJK_bs.reset();
+
+    //Konrad - all the stuff above still needed but I adjusted the scale factors in \omega's
+    //we don't need the DF stuff that was below
+    
+    double CompleteInd30 = 0.0;
+  
+    auto sAR = std::make_shared<Matrix>("sAR", na, nr); 
+    double **sARp = sAR->pointer();
+
+    for (int a = 0; a < na; a++) {
+        for (int r = 0; r < nr; r++) {
+            sARp[a][r] = wBAR_[a][r] / (evalsA_[a] - evalsA_[na+r]);
+        }
+    }
+
+    auto sBS = std::make_shared<Matrix>("sBS", nb, ns); 
+    double **sBSp = sBS->pointer();
+
+    for (int b = 0; b < nb; b++) {
+        for (int s = 0; s < ns; s++) {
+            sBSp[b][s] = wABS_[b][s] / (evalsB_[b] - evalsB_[nb+s]);
+        }
+    }
+
+    std::shared_ptr<Matrix> STS_br = linalg::triplet(sBS, Tas, sAR, false, true, false);
+    std::shared_ptr<Matrix> STS_as = linalg::triplet(sAR, Tbr, sBS, false, true, false);
+    std::shared_ptr<Matrix> STS_ar = linalg::triplet(sAR, Tar, sAR, false, true, false);
+    std::shared_ptr<Matrix> STS_bs = linalg::triplet(sBS, Tbs, sBS, false, true, false);
+
+    CompleteInd30 += C_DDOT(na * nr, uAR[0], 1, omega_ar->pointer()[0], 1);
+    CompleteInd30 += C_DDOT(nb * ns, uBS[0], 1, omega_bs->pointer()[0], 1);
+    CompleteInd30 -= C_DDOT(nb * nr, STS_br->pointer()[0], 1, omega_br->pointer()[0], 1);
+    CompleteInd30 -= C_DDOT(na * ns, STS_as->pointer()[0], 1, omega_as->pointer()[0], 1);
+    CompleteInd30 -= C_DDOT(na * nr, STS_ar->pointer()[0], 1, omega_ar->pointer()[0], 1);
+    CompleteInd30 -= C_DDOT(nb * ns, STS_bs->pointer()[0], 1, omega_bs->pointer()[0], 1);
+
+//All Omega-dependent contributions have been completed, now Xi-dependent (J/K) contributions
+
+    auto SCt_Na = std::make_shared<Matrix>("SCt_Na", nn, na);
+    auto SCt_Nb = std::make_shared<Matrix>("SCt_Nb", nn, nb);
+    C_DGEMM('N', 'T', nn, na, nr, 1.0, Ct_Kr->pointer()[0], nr, sAR->pointer()[0], nr, 0.0,
+            SCt_Na->pointer()[0], na);
+    C_DGEMM('N', 'T', nn, nb, ns, 1.0, Ct_Ks->pointer()[0], ns, sBS->pointer()[0], ns, 0.0,
+            SCt_Nb->pointer()[0], nb);
```

Comment:
I believe you can just hit these with doublet?

Reply:
Yes, I can, and now I do.